routes.go

```diff
@@ -441,6 +444,88 @@ func deleteTribe(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(true)
 }
 
+func getChannelsByTribe(w http.ResponseWriter, r *http.Request) {
+	all := r.URL.Query().Get("all")
+	channels := []Channel{}
+	tribeUUID := chi.URLParam(r, "tribeUUID")
+	if all == "true" {
+		channels = DB.getAllChannelsByTribe(tribeUUID)
+	} else {
+		channels = DB.getChannelsByTribe(tribeUUID)
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(channels)
+}
+
+func createOrEditChannel(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	pubKeyFromAuth, _ := ctx.Value(ContextKey).(string)
+
+	channel := Channel{}
+	body, err := ioutil.ReadAll(r.Body)
+	r.Body.Close()
+	err = json.Unmarshal(body, &channel)
+	if err != nil {
+		fmt.Println(err)
+		w.WriteHeader(http.StatusNotAcceptable)
+		return
+	}
+
+	if channel.TribeUUID == "" {
+		fmt.Println("createOrEditChannel no tribe uuid")
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	now := time.Now() //.Format(time.RFC3339)
+
+	extractedPubkey, err := VerifyTribeUUID(channel.TribeUUID, false)
+	if err != nil {
+		fmt.Println(err)
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	if pubKeyFromAuth == "" {
+		channel.Created = &now
+	} else { // IF PUBKEY IN CONTEXT, MUST AUTH!
+		if pubKeyFromAuth != extractedPubkey {
+			fmt.Println("createOrEditChannel pubkeys dont match")
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+	}
+
+	/*
+			existing := DB.getTribe(channel.TribeUUID)
```

Comment:
Do I need to generate a unique `id` for the channel like we do in the `createOrEditTribe` function for `uuid`

Reply:
That createOrEditTribe function is complex for a variety of reasons, you dont need to do that for Channels.

```
func (db database) createChannel(c Channel) Channel {
	db.db.Create(&c)
	return c
}
func (db database) updateChannelName(id uint, name string) {
  if id==0 return // validate theres really an id
  db.Model(&Channel{}).Where("id = ?", id).Update("name", name)
}
```

Reply:
Yes, channels should have unique ID (auto-incrementing INT)

```sql
CREATE TABLE people (
  id SERIAL PRIMARY KEY,
  tribe_uuid TEXT NOT NULL,
  name TEXT NOT NULL,
  created timestamptz,
);
```